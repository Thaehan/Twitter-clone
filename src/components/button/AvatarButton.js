import { View } from 'react-native';
import React from 'react';
import {
  TouchableOpacity,
  StyleSheet,
  Image,
} from 'react-native';
import { HEADER_HEIGHT } from '../../styles/Style';

//Avatar button, chuyển người dùng đến trang của người dùng đã được chọn, không chuyển nếu screen hiện tại đang là của người đó
export default function AvatarButton({
  style,
  userId,
  source,
  size,
  onPress,
}) {
  const redirectToUserProfileScreen = (id) => {};
  return (
    <View style={style}>
      <TouchableOpacity
        style={styles.container}
        onPress={onPress}
      >
        <Image
          source={{ uri: source }}
          style={{
            height: size,
            width: size,
            borderRadius: 50,
          }}
          testID="AvatarImage"
        />
      </TouchableOpacity>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    alignContent: 'center',
    alignSelf: 'center',
  },
});
