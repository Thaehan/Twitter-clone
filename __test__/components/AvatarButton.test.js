import React from 'react';
// import renderer from 'react-test-renderer';\
import { render } from '@testing-library/react-native';
import { Provider } from 'react-redux';

import store from '../../src/redux/store';
import AvatarButton from '../../src/components/button/AvatarButton';

jest.useFakeTimers();
jest.mock(
  'react-native/Libraries/Animated/NativeAnimatedHelper'
);

describe('<Login />', () => {
  it('Render avatarButton', () => {
    const testComponent = render(<AvatarButton />);
    expect(testComponent.toJSON()).toMatchSnapshot();
    // testComponent.getByTestId('AvatarImage').tobe();
  });
});
