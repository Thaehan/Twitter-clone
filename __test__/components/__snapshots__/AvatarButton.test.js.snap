// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Render avatarButton 1`] = `
<View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignContent": "center",
        "alignSelf": "center",
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Object {
          "borderRadius": 50,
          "height": undefined,
          "width": undefined,
        }
      }
      testID="AvatarImage"
    />
  </View>
</View>
`;

exports[`<Login /> Render avatarButton 1`] = `
<View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignContent": "center",
        "alignSelf": "center",
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Object {
          "borderRadius": 50,
          "height": undefined,
          "width": undefined,
        }
      }
      testID="AvatarImage"
    />
  </View>
</View>
`;
